lib/Sema/CSSimplify.cpp

```diff
@@ -3534,6 +3534,28 @@ ConstraintSystem::matchTypes(Type type1, Type type2, ConstraintKind kind,
       if (result.isSuccess() || !shouldAttemptFixes())
         return result;
     }
+
+    auto last = locator.last();
+    if (last && last->is<LocatorPathElt::ExplicitTypeCoercion>()) {
+      bool hasDeepEqualityRestriction = llvm::any_of(
+          conversionsOrFixes, [](const RestrictionOrFix &entry) -> bool {
+            return entry.IsRestriction &&
+                   entry.getRestriction() ==
+                       ConversionRestrictionKind::DeepEquality;
+          });
+
+      // If there is a deep equality restriction already recorded
+      // or the toType of the coercion is a existential,
+      // let's skip this force downcast here because the actual correct
+      // fix is going to be added later.
+      if (!hasDeepEqualityRestriction && !type2->isExistentialType()) {
```

Comment:
What is the correct correct fix here? Once this is in `repairFailures` you can use `if (hasRestrictionOrConversion(ConversionRestrictionKind::Existential)) break;` instead because conversion to existential type would record `Existential` restriction.